src/asset/asset-registry.js

```diff
@@ -220,6 +220,7 @@ pc.extend(pc, function () {
         remove: function (asset) {
             delete this._cache[asset.id];
             delete this._names[asset.name];
+            this._assets.splice(this._assets.indexOf(asset), 1);
```

Comment:
This is dangerous if the asset does not exist. `indexOf` would return `-1` and this would happen:

```js
var a = [1,2,3];
a.splice(-1, 1);
console.log(a);
```

prints:

> [1, 2]

Reply:
I agree.

Reply:
@ronkorving You are right.

Reply:
Done.